source/common/http/conn_manager_impl.cc

```diff
@@ -140,7 +140,9 @@ void ConnectionManagerImpl::doEndStream(ActiveStream& stream) {
   // Reading may have been disabled for the non-multiplexing case, so enable it again.
   if (drain_state_ != DrainState::Closing && codec_->protocol() != Protocol::Http2 &&
       !read_callbacks_->connection().readEnabled()) {
-    read_callbacks_->connection().readDisable(false);
+    while (!read_callbacks_->connection().readEnabled()) {
```

Comment:
You're totally right - I found this call site first and newStream later.  reverted.